WebInstall/explainer_cross_domain.md

```diff
@@ -0,0 +1,259 @@
+
+### **Web app installation from associated domain**
+
+An (out-of-scope) origin associated to a web app could prompt for the installation of said web app. The typical use case for this is a marketing website that can offer their customers a direct way to install the associated web app.
+
+As an example, a user can navigate to the `streamflix.com` website and find UX that allows them to install the associated application found in `streamflix.app`.
+
+```javascript
+/* Example that uses the Permissions API to check if the installation permission is set before calling the install method */
+  const installApp = async () => {
+    try{
+        const { state } = await navigator.permissions.query({
+            name: "installation"
+          });
+          switch (state) {
+            case "granted":
+                const value = await navigator.install('https://streamflix.app');
+              break;
+            case "prompt":
+              showInstallButton();
+              break;
+            case "denied":
+              browseToAppStorePage();
+              break;
+          }
+    }
+    catch(err){console.error(err.message)}
+  };
+```
+
+Manifest file for the `streamflix.app`, allowing installation *ONLY* from `streamflix.com` :
+
+```json
+{
+    "name": "Streamflix App",
+    "display": "standalone",
+    "start_url": "/index.html",
+    "install_sources": [ 
+	    {"origin": "streamflix.com"}   
+    ]
+}
+```
+
+![Different domain install flow](./difdomaininstall.png) 
+
+### **Creation of online catalogs**
+
+ The other use case for the API is related to the creation of online catalogs. A web site/app can list and install web apps. For example, `store.com` would be able to distribute apps on multiple platforms and multiple devices.
+
+```javascript
+/* tries to install a cross-origin web app */
+
+const installApp = async (manifest_id) => {
+    if ('install' in navigator === false) return; // api not supported
+    try {
+            await navigator.install(manifest_id);
+    } catch(err) {
+        switch(err.message){
+            case 'AbortError':
+                /* Operation was aborted*/
+                break;
+           
+        }
+    }
+};
+```
+
+  ![Install flow from an app repository](./apprepositoryinstallation.png) 
+
+## Proposed Solution
+
+### The `navigator.install` method
+
+To install a web app, a web site would use the promise-based method `navigator.install([<manifest_id>[, <params>]]);`. This method will:
+
+* Resolve when an installation was completed.
+    * The success value will be an object that contains:
+     	*  `mode`: string with the surface-hint where the app was installed.
+* Be rejected if the prompt is not shown or if the app installation did not complete. It'll reject with a [`DOMException`](https://developer.mozilla.org/en-US/docs/Web/API/DOMException) value of:
+    * `AbortError`: The installation (prompt) was closed/cancelled.
```

Comment:
WDYT about a `ManifestIdMismatchError` (or something more concise) to indicate that the install failed because the specified `manifest_id` did not match the id read from the manifest at the `install_url`?